core/services/pipeline/models.go

```diff
@@ -66,12 +70,7 @@ func (r *Run) SetID(value string) error {
 }
 
 func (r Run) HasErrors() bool {
-	return r.FinalErrors().HasErrors()
-}
-
-func (r Run) FinalErrors() (f FinalErrors) {
-	f, _ = r.Errors.Val.(FinalErrors)
-	return f
+	return r.Errors.Val != nil && !r.Errors.Null
```

Comment:
Is this definitely correct?

I believe it used to do something different. See: https://github.com/smartcontractkit/chainlink/pull/4110/files#diff-fa579cc84242c95b0c034798c92c8f78b9f515db2f884e985d0ee650807a5672L56

Reply:
If the errors val is `[null]` then it should NOT have errors

Reply:
spoke offline - added a custom error type